### PR TITLE
Dynamically add configuration for GCP 

### PR DESCRIPTION
**Pull request by cassand**
This mechanism could be used to update server.xml based on environment variables #57

### CODE REVIEW

File: tomcat/pom.xml [hunk 1]
```diff
@@ -183,6 +183,25 @@
...
+            <id>copy-tomcat-optional-configuration</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}/docker-src/config</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>${project.basedir}/src/main/optional-config</directory>
```
Comment (meltsufin) [inline tomcat/pom.xml]: Should this be in `/src/main/resources`?

File: tomcat/src/main/optional-config/gcp.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<Valve className="org.apache.catalina.valves.RemoteIpValve"
+remoteIpHeader="x-forwarded-for"
+protocolHeader="x-forwarded-proto"
+trustedProxies=".*"
```
Comment (meltsufin) [inline tomcat/src/main/optional-config/gcp.xml]: Can we use externalized `{properties}` here?

File: tomcat/src/main/tomcat-base/conf/server.xml [hunk 3]
```diff
@@ -17,6 +17,7 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 -->
+<!DOCTYPE server-xml [<!ENTITY gcp-configuration SYSTEM "gcp.xml">]>
```
Comment (meltsufin) [inline tomcat/src/main/tomcat-base/conf/server.xml:20]: What happens when `gcp.xml` is empty?

Comment (cassand) [reply tomcat/src/main/tomcat-base/conf/server.xml:20]: If `gcp.xml` is empty then nothing is added to the configuration (gcp valves). If `gcp.xml` is missing then an error is thrown when the server is started.

Comment (meltsufin) [reply tomcat/src/main/tomcat-base/conf/server.xml:20]: OK. I think this is the kind of thing that calls for tests.

File: tomcat/src/main/tomcat-base/conf/catalina.properties [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+log4j-web*.jar,log4j-taglib*.jar,log4javascript*.jar,slf4j-taglib*.jar
+
+# String cache configuration.
+tomcat.util.buf.StringCache.byte.enabled=true
+#tomcat.util.buf.StringCache.char.enabled=true
+#tomcat.util.buf.StringCache.trainThreshold=500000
+#tomcat.util.buf.StringCache.cacheSize=5000
+
+# Regular expression to indicate witch proxy should be trusted in x-forwarded-for header (e.g load balancer)
+gae.proxy.trusted=".*"
```
Comment (meltsufin) [inline tomcat/src/main/tomcat-base/conf/catalina.properties:151]: Before you added this file, were none of the above defaults applied?
Also, we should start working on #56 after this is merged.

Comment (cassand) [reply tomcat/src/main/tomcat-base/conf/catalina.properties:151]: This configuration file was copied from Tomcat Home during the docker build, the defaults will remain the same.

**Review by meltsufin — DISMISSED** [review]

**Review by meltsufin — APPROVED** [review]

**Pull request merged**